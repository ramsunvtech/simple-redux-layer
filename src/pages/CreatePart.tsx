import React from 'react';

const CreatePart = () => <p>Create part form should be here</p>;

export default CreatePart;
