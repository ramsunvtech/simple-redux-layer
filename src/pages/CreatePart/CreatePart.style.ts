import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  nameField: {
    margin: theme.spacing(0, 0, 2, 0),
  },
  statusField: {
    margin: theme.spacing(0, 0, 2, 0),
  },
}));