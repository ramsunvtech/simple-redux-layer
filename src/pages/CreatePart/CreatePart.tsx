import React, { useState } from 'react';

// `Hooks`.
import { useHistory } from "react-router-dom";

// `Material` Components.
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import TextField from '@material-ui/core/TextField';
import Switch from '@material-ui/core/Switch';
import Button from '@material-ui/core/Button';

// Tiny Redux.
import { tinyConnect } from '../../lib/tinyReactRedux';

// `Actions`.
import * as actionTypes from '../../store/actions/actionTypes';

// Stylesheet.
import useStyles from './CreatePart.style';

// `Models`.
import Part from '../../models/Part';

interface CreatePartPageProps {
  createPart: (part: Part) => void;
};

export const CreatePart: React.FC<CreatePartPageProps> = ({ createPart }) => {
  const classes = useStyles();
  const history = useHistory();
  const [checkStatus, setCheckStatus] = useState(true);
  const [partName, setPartName] = useState('');

  const handleSwitch = (event: React.ChangeEvent<HTMLInputElement>) => {
    setCheckStatus(!checkStatus);
  };

  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setPartName(event.target.value);
  };

  const onSubmit = (e: React.FormEvent) => {
    e.preventDefault();

    createPart({
      name: partName, id: Date.now(), status: checkStatus
    });
    history.push('/');
  };

  return (
    <form noValidate autoComplete="off" onSubmit={onSubmit}>
      <Grid container>
        <Grid item xl={12}>
          <Typography variant="h4" gutterBottom>
            Create Part
          </Typography>
        </Grid>
        <Grid item xl={5} sm={12} className={classes.nameField}>
          <TextField
            id="name"
            label="Name"
            value={partName}
            InputProps={{
              style: {
                color: 'black'
              }
            }}
            color="secondary"
            onChange={handleNameChange}
            fullWidth
          />
        </Grid>
        <Grid item xl={12} sm={12} className={classes.statusField}>
          <FormControlLabel
            control={
              <Switch
                checked={checkStatus}
                onChange={handleSwitch}
                color="primary"
                name="switch"
                inputProps={{ 'aria-label': 'primary checkbox' }}
              />
            }
            label="Status"
          />
        </Grid>
        <Grid item xl={3} sm={12}>
          <Button type="submit" variant="contained" color="primary" fullWidth>
            Create Part
          </Button>
        </Grid>
      </Grid>
    </form>
  );
};

const mapDispatchToProps = (dispatch: Function) => {
  return {
    createPart: (partItem: Part) => dispatch({ type: actionTypes.CREATE_PART, partItem }),
  }
}

export default tinyConnect(() => {}, mapDispatchToProps)(CreatePart);