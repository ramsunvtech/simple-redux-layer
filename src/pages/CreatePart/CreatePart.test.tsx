import React from 'react';
import {
  render,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// `Test` Component.
import { CreatePart } from './CreatePart';

jest.mock('../../lib/tinyReactRedux', () => {
  return {
    tinyConnect: (a, b) => (Component:any) => {
      return <Component createPart={() => {}} />
    },
  };
});

describe('CreatePart', () => {
  afterEach(cleanup);

  it('should take a snapshot', () => {
    const { asFragment } = render(<CreatePart />);
    expect(asFragment(<CreatePart />)).toMatchSnapshot();
  });
});
