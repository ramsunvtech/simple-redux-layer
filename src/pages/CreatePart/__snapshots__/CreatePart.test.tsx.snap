// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePart should take a snapshot 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    novalidate=""
  >
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xl-12"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Create Part
        </h4>
      </div>
      <div
        class="MuiGrid-root makeStyles-nameField-1 MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-xl-5"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="name"
            id="name-label"
          >
            Name
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            style="color: black;"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-statusField-2 MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-xl-12"
      >
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            class="MuiSwitch-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-4 Mui-checked"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-label="primary checkbox"
                  checked=""
                  class="PrivateSwitchBase-input-6 MuiSwitch-input"
                  name="switch"
                  type="checkbox"
                  value=""
                />
                <span
                  class="MuiSwitch-thumb"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Status
          </span>
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-xl-3"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Create Part
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
